src/java.base/share/classes/jdk/internal/vm/VMSupport.java

```diff
@@ -577,4 +577,20 @@ private static int readLength(DataInputStream dis) throws IOException {
         }
         return length;
     }
+
+    /**
+     * Allocate an array of a value class type with components that behave in
+     * the same way as a {@link jdk.internal.vm.annotation.NullRestricted}
+     * field.
+     * <p>
+     * Because these behaviors are not specified by Java SE, arrays created with
+     * this method should only be used by internal JDK code for experimental
+     * purposes and should not affect user-observable outcomes.
+     *
+     * @throws IllegalArgumentException if {@code componentType} is not a
+     *         value class type or is not annotated with
+     *         {@link jdk.internal.vm.annotation.ImplicitlyConstructible}
+     */
+    public static native Object[] newNullRestrictedArray(Class<?> componentType,
```

Comment:
I think this API should be moved to some other runtime class.    VMSupport is the support class used by JVM CI, JVM TI, and attach mechanism.

What about `jdk.internal.value.ValueClassSupport` or `jdk.internal.misc.VM` for now?   We can find a better home later.